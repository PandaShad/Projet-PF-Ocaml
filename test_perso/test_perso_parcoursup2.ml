open Test_perso_parcoursup_utils

let session = nouvelle_session ()

let () = 
    ajoute_candidat session ~nom_candidat:"Maxime";
    ajoute_candidat session ~nom_candidat:"Ghouti";
    ajoute_candidat session ~nom_candidat:"Rami";

    ajoute_formation session ~nom_formation:"Université Côte d'Azur" ~capacite:2;
    ajoute_formation session ~nom_formation:"Université de Toulon" ~capacite:1;

    ajoute_voeu session 
        ~rang_repondeur: None 
        ~nom_candidat:"Maxime" 
        ~nom_formation:"Université Côte d'Azur";
    ajoute_voeu session 
        ~rang_repondeur: (Some 1) 
        ~nom_candidat:"Maxime" 
        ~nom_formation:"Université de Toulon";
    ajoute_voeu session 
        ~rang_repondeur:(Some 2) 
        ~nom_candidat:"Ghouti" 
        ~nom_formation:"Université Côte d'Azur";
    ajoute_voeu session 
        ~rang_repondeur: (Some 1) 
        ~nom_candidat:"Ghouti" 
        ~nom_formation:"Université de Toulon";
    ajoute_voeu session 
        ~rang_repondeur:(Some 1) 
        ~nom_candidat:"Rami" 
        ~nom_formation:"Université Côte d'Azur";
    ajoute_voeu session 
        ~rang_repondeur: (Some 1) 
        ~nom_candidat:"Rami" 
        ~nom_formation:"Université de Toulon";
        
    ajoute_commission session 
        ~nom_formation:"Université Côte d'Azur" 
        ["Ghouti";"Maxime";"Rami"]; 

    ajoute_commission session 
        ~nom_formation:"Université de Toulon" 
        ["Ghouti";"Maxime";"Rami"]; 

    reunit_commissions session;

    nouveau_jour session;
    affiche_voeux_en_attente session "Maxime";
    affiche_propositions_en_attente session "Maxime";
    affiche_voeux_en_attente session "Ghouti";
    affiche_propositions_en_attente session "Ghouti";
    affiche_voeux_en_attente session "Rami";
    affiche_propositions_en_attente session "Rami";
    (* renonce session 
        ~nom_candidat:"Maxime" 
        ~nom_formation:"Université Côte d'Azur"; *)
    renonce session 
        ~nom_candidat:"Ghouti" 
        ~nom_formation:"Université de Toulon";

    nouveau_jour session;
    affiche_voeux_en_attente session "Maxime";
    affiche_propositions_en_attente session "Maxime";
    affiche_voeux_en_attente session "Ghouti";
    affiche_propositions_en_attente session "Ghouti";
    affiche_voeux_en_attente session "Rami";
    affiche_propositions_en_attente session "Rami";