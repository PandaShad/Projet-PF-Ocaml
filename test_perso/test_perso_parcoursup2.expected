nouvelle session
ajoute candidat Maxime
ajoute candidat Ghouti
ajoute candidat Rami
ajoute formation Université Côte d'Azur (2 places)
ajoute formation Université de Toulon (1 places)
ajoute voeu de Maxime pour Université Côte d'Azur
ajoute voeu de Maxime pour Université de Toulon (rang=1)
ajoute voeu de Ghouti pour Université Côte d'Azur (rang=2)
ajoute voeu de Ghouti pour Université de Toulon (rang=1)
ajoute voeu de Rami pour Université Côte d'Azur (rang=1)
ajoute voeu de Rami pour Université de Toulon (rang=1)
commission de classement de Université Côte d'Azur : [Ghouti; Maxime; Rami]
commission de classement de Université de Toulon : [Ghouti; Maxime; Rami]
réunit commissions
nouveau jour
Voeux en attente de Maxime : {Université de Toulon(0)}
Propositions en attente de Maxime : {Université Côte d'Azur}
Voeux en attente de Ghouti : {}
Propositions en attente de Ghouti : {Université de Toulon}
Voeux en attente de Rami : {Université Côte d'Azur(0), Université de Toulon(1)}
Propositions en attente de Rami : {}
Maxime renonce à Université Côte d'Azur
Ghouti renonce à Université de Toulon
nouveau jour
Voeux en attente de Maxime : {}
Propositions en attente de Maxime : {Université de Toulon}
Voeux en attente de Ghouti : {}
Propositions en attente de Ghouti : {}
Voeux en attente de Rami : {Université de Toulon(0)}
Propositions en attente de Rami : {Université Côte d'Azur}
